views/survey.pug

```diff
@@ -17,7 +17,7 @@ html
           ol
             each item in survey
               if item.prompt
-                if item.group_prompt
+                if item.group_prompt !== 'undefined'
```

Comment:
I'd do this the other way around. Check if item.group_prompt is equal to the previous item.group_prompt. If it is, show nothing, if not show the new prompt. 

This way you handle any kind of item group prompt differences etc. 

Also, if its blank perhaps say something like "Answer the following questions" (which will appear at the top for any survey that has no group_prompt.